### PR TITLE
Report POH in ETW events

### PR DESCRIPTION
**Pull request by VSadov**
Follow-up item for: https://github.com/dotnet/runtime/issues/13731

=== When existing events have a way to specify generation, just report POH.
- [x] GCCreateSegment_V1
- [x] GCAllocationTick_V3
- [x] GCGenerationRange (changes in `descr_generations_to_profiler`)

=== Extend existing events to fit POH.
- [x]  BGCOverflow
- [x]  GCHeapStats_V1

### CODE REVIEW

**Comment by VSadov** [comment]
Made new ETW event versions backwards compatible. Now this is ready for review.

**Comment by VSadov** [comment]
Created an issue in perfview repo to add support for this on PerfView side once this PR is merged. - https://github.com/microsoft/perfview/issues/1146

File: src/coreclr/src/gc/gcee.cpp [hunk 1]
```diff
@@ -401,12 +417,18 @@ void GCHeap::DiagTraceGCSegments()
...
+        for (int i = uoh_start_generation; i < total_generation_count; i++)
         {
-            uint8_t* address = heap_segment_mem (seg);
-            size_t size = heap_segment_reserved (seg) - heap_segment_mem (seg);
-            FIRE_EVENT(GCCreateSegment_V1, address, size, static_cast<uint32_t>(gc_etw_segment_large_object_heap));
+            for (seg = generation_start_segment (h->generation_of (i)); seg != 0; seg = heap_segment_next(seg))
+            {
+                uint8_t* address = heap_segment_mem (seg);
+                size_t size = heap_segment_reserved (seg) - heap_segment_mem (seg);
+                gc_etw_segment_type segment_type = (i == 3) ?
```
Comment (Maoni0) [inline src/coreclr/src/gc/gcee.cpp:427]: > gc_etw_segment_type segment_type = (i == 3) ? [](start = 15, length = 46)

why not do (i == loh_generation) here?

**Review by Maoni0 — CHANGES_REQUESTED** [review]
the rest LGTM! do you also have the change in TraceEvent for this (to test)?

File: src/coreclr/src/gc/gcee.cpp (hunk 1 above)
Comment (VSadov) [reply src/coreclr/src/gc/gcee.cpp:427]: Just did not think about it. It does seem odd.

**Review by Maoni0 — APPROVED** [review]
LGTM!

**Comment by VSadov** [comment]
Thanks!!!

**Pull request merged**